packages/firestore/test/unit/local/persistence_test_helpers.ts

```diff
@@ -44,3 +55,42 @@ export async function testMemoryPersistence(): Promise<MemoryPersistence> {
   await persistence.start();
   return persistence;
 }
+
+/**
+ * Creates LocalStorageNotificationChannel instance for testing, destroying any
+ * previous contents if they existed.
+ */
+export async function testLocalStorageNotificationChannel(
+  ownerId: string,
+  existingMutationBatches: BatchId[],
+  existingQueryTargets: TargetId[]
+): Promise<TabNotificationChannel> {
+  window.localStorage.clear();
+  const instances = [];
+
+  if (existingMutationBatches.length + existingQueryTargets.length != 0) {
+    const secondaryChannel = new LocalStorageNotificationChannel(
```

Comment:
This feels a bit dirty, I'd add a comment.

// HACK: Create a secondary channel to seed the data into LocalStorage. 
// NOTE: We don't call shutdown() on it because that would delete the data.

Reply:
Done